### PR TITLE
(frontend) set up and adopt ESLint

### PR DESCRIPTION
**Pull request by cloudyyoung**
ESLint and Prettier can work together and take care of code smells and format, respectively. Here's an article: https://www.robinwieruch.de/prettier-eslint/

To have extensive ESLint integration in vs code, install this plugin: https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint. On a file, press F1 and run "ESLint: Fix all auto-fixable problems".

Or simply command line `npm run lint`

Finding code smells:
image

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -15,7 +15,6 @@
     "@types/react": "^18.0.24",
     "@types/react-dom": "^18.0.8",
     "axios": "^1.1.3",
-    "prettier": "^2.7.1",
```
Comment (cloudyyoung) [inline client/package.json:18]: `prettier` should live in `devDependencies` instead of `dependencies`

File: client/package.json [hunk 2]
```diff
@@ -53,7 +45,13 @@
     ]
   },
   "devDependencies": {
+    "@typescript-eslint/eslint-plugin": "^5.45.1",
+    "@typescript-eslint/parser": "^5.45.1",
     "autoprefixer": "^10.4.13",
+    "eslint": "^8.29.0",
+    "eslint-config-prettier": "^8.5.0",
+    "eslint-plugin-prettier": "^4.2.1",
```
Comment (cloudyyoung) [inline client/package.json:53]: As a form of ESLint plugin, `prettier` standalone lib is not imported

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -33,7 +33,7 @@ jobs:
         run: |
           cd client
           npm install
-          npm run check-format
+          eslint . --ext .js,.jsx,.ts,.tsx
```
Comment (tim-macphail) [inline .github/workflows/ci.yaml:36]: Right now this is failing 
image

File: client/package.json [hunk 4]
```diff
@@ -30,16 +29,9 @@
     "start": "react-scripts start",
     "build": "react-scripts build",
     "test": "react-scripts test",
-    "format": "prettier --write .",
-    "check-format": "prettier --check .",
+    "lint": "eslint . --ext .js,.jsx,.ts,.tsx --fix",
```
[the remark this answers is not in the document]
Comment (cloudyyoung) [reply client/package.json:33]: Should be good now

**Review by tim-macphail — APPROVED** [review]

**Review by BradleyLi — APPROVED** [review]
Everything is good!

**Pull request merged**